react/week4/github_users_search/src/index.js

```diff
@@ -0,0 +1,146 @@
+    return fetch(URL).then((response) => {
+      if (response.status >= 200 && response.status <= 299) {
+        return response.json();
+      } else {
+        throw Error(response.statusText);
+      }
+    })
+  }
+}
+
+function fetchGithubUsersRepos(login) {
+  const URL = `https://api.github.com/users/${login}/repos`
+  return fetch(URL).then((response) => {
+    if (response.status >= 200 && response.status <= 299) {
+      return response.json();
+    } else {
+      throw Error(response.statusText);
+    }
+  })
+}
+
+function Header() {
+  return (
+    <div>
+      <h1>Github user searcher</h1>
+    </div>
+  )
+}
+
+class Searchbar extends React.Component {
+  state = {
+    searchTerm: '',
+    isLoading: false,
+    usersArray: [],
+    reposArray: [],
+    errorMessage: '',
+    isClickedUser: false,
+    isClickedRepos: false,
+    selectedUser: ''
+  }
+
+  handleSearchChange = (event) => {
+    const searchedValue = event.target.value
+    this.setState({
+      searchTerm: searchedValue
+    })
+    if (searchedValue !== '') {
+      fetchGithubUsers(searchedValue)
+        .then(data => {
+          console.log(data);
+          const users = data.items.map(user => user)
+          console.log(users);
+          this.setState({
+            usersArray: users,
+            isLoading: true
+          })
+        }).catch(error => {
+          console.log(error.message);
+          this.setState({
+            errorMessage: error.message,
+            usersArray: []
+          })
+        }).finally(() => this.setState({ isLoading: false }))
+    }
+  }
+
+  handleClickSearch = (event, loginName) => {
+    event.preventDefault();
+    console.log(loginName);
+    fetchGithubUsersRepos(loginName)
+      .then(data => {
+        const repos = data.map(repo => repo)
+        console.log(repos)
+        this.setState({
+          reposArray: repos,
+          isClickedUser: true,
+          selectedUser: loginName,
+          searchTerm: loginName
+        })
+      })
+  }
+
+  handleClickRepos = (event) => {
+    event.preventDefault();
+    this.setState({
+      isClickedRepos: true
+    })
+  }
+
+  render() {
+    const contextValue = {
+      usersArray: this.state.usersArray,
+      isClickedUser: this.state.isClickedUser,
+      isClickedRepos: this.state.isClickedRepos,
+      reposArray: this.state.reposArray,
+      handleClickSearch: this.handleClickSearch,
+      handleClickRepos: this.handleClickRepos,
+      loginName: this.state.selectedUser
+    }
+    return (
+      <ContextProvider value={contextValue}>
+        <div className='list-container'>
+          <input type='text' placeholder='Search for user' value={this.state.searchTerm} onChange={this.handleSearchChange} />
+          {this.state.isLoading ? <p>Loading...</p> : (this.state.usersArray.length === 0 || this.state.searchTerm === '') ? <p>No results</p> :
+            <div className='list-wrapper'>
+              <ul>
```

Comment:
Since the component `UserList` is, by its name and content, a list, I would move `<ul>` to inside that component, so that it ends up rendering something like
```
<div>
  <ul>
     <li>...</li>
   </ul>
</div>
```